agent/replacecall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -0,0 +1,398 @@
+    }
+
+    try {
+      ClassGen cg = new ClassGen(c);
+      if (transformClass(cg)) {
+        JavaClass javaClass = cg.getJavaClass();
+        if (ReplaceCallAgent.debug) {
+          Path filepath = ReplaceCallAgent.debugPath.resolve(className + ".class");
+          javaClass.dump(filepath.toFile());
+        }
+        debug_transform.log("transform: EXIT class %s transformed", className);
+        return javaClass.getBytes();
+      } else {
+        debug_transform.log("transform: EXIT class %s not transformed", className);
+        return null;
+      }
+    } catch (Throwable e) {
+      debug_transform.log(
+          "transform: EXIT transform of %s resulted in exception %s%n", className, e);
+      System.out.format(
+          "Unexpected exception %s (%s) in class transform of %s%n", e, e.getCause(), className);
+      e.printStackTrace();
+      return null;
+    }
+  }
+
+  /**
+   * Indicates whether the class is a boot-loaded class.
+   *
+   * <p>Checks whether the class is loaded by the bootstrap loader or by the first classloader. The
+   * first classloader will either be a user-provided boot loader, or the extension class loader.
+   * Since the predicate is for performance, we don't make the extra check to determine if the user
+   * has given a boot loader.
+   *
+   * @param loader the classloader for the class
+   * @return true if the named class is boot-loaded, false otherwise
+   */
+  private boolean isBootloadedClass(ClassLoader loader) {
+    return loader == null || loader.getParent() == null;
+  }
+
+  /**
+   * Indicates whether the class is in the {@code java.awt.} or {@code javax.swing.} packages.
+   *
+   * @param fullClassName the fully-qualified class name
+   * @return true if the class is in {@code java.awt.} or {@code javax.swing.}, false otherwise
+   */
+  private boolean isGUIClass(String fullClassName) {
+    return fullClassName.startsWith("java.awt.") || fullClassName.startsWith("javax.swing.");
+  }
+
+  /**
+   * Indicates whether the named class is defined in a package that is excluded.
+   *
+   * @param fullClassName the fully-qualified class name, must be non-null
+   * @return true if any excluded package is a prefix of the class name, false otherwise
+   */
+  private boolean isExcludedClass(ClassLoader loader, String fullClassName) {
+    // For performance, skip boot loaded classes, but include GUI classes
+    if (isBootloadedClass(loader) && !isGUIClass(fullClassName)) {
+      return true;
+    }
+
+    for (String prefix : excludedPackagePrefixes) {
+      if (fullClassName.startsWith(prefix)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Processes each method in the given class replacing any specified calls. The replacements are
+   * static methods.
+   *
+   * @param cg the BCEL class representation
+   * @return true if the class has been transformed, false if either no calls were replaced or an
+   *     error occured
```

Comment:
This is ambiguous for callers.  It's also error-prone:  the implementation might not output about errors.
I suggest that the method contract be changed so null only means no calls were replaced.